gremlin-server/src/main/java/org/apache/tinkerpop/gremlin/server/op/AbstractEvalOpProcessor.java

```diff
@@ -231,9 +233,18 @@ protected void evalOpInternal(final Context ctx, final Supplier<GremlinExecutor>
         final GremlinExecutor.LifeCycle lifeCycle = GremlinExecutor.LifeCycle.build()
                 .evaluationTimeoutOverride(seto)
                 .afterFailure((b,t) -> {
+                    graphManager.onQueryError(msg, t);
                     if (managedTransactionsForRequest) attemptRollback(msg, ctx.getGraphManager(), settings.strictTransactionManagement);
                 })
+                .afterTimeout(b -> {
```

Comment:
this is unfortunate. i assume there is no way to get the original time-out exception thrown? do we really have to construct a fresh exception here just for purpose of getting one to `onQueryError()`?

Reply:
sorry for the delay in reviewing your latest changes. unfortunately, i think the breaking change forces this into 3.7.x rather than 3.6.x. there are several providers who rely on this API. Maybe you could turn it back into a non-breaking change though? Since it is just a callback, you could add the new signature for `afterTimeout()` as an overload. You could then have `GremlinExecutor` call both. Deprecate the `Consumer` version and ensure the javadoc makes clear that users should only utilize one of those two methods as a single timeout triggers them both. Would that work?

Reply:
@spmallette In https://github.com/apache/tinkerpop/pull/1837/commits/718c0830d0725655b8c5999832e12f917991a9fc I deprecated the use of `afterTimeout` as a consumer and I had the deprecated method create a new `BiConsumer` which calls the updated `afterTimeout` method. I also reverted the `GremlinExecutor` tests to use the deprecated method to make sure they are still working, I can either move them to the new method or leave things as they are